pkg/kbld/config/image_meta.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"encoding/json"
+
+	"sigs.k8s.io/yaml"
+)
+
+type Meta interface {
+	meta()
+}
+
+type ImageMeta struct {
+	Metas []Meta
+}
+
+type BuiltImageSourceGit struct {
+	Type      string // always set to 'git'
+	RemoteURL string `json:",omitempty" yaml:",omitempty"`
+	SHA       string
+	Dirty     bool
+	Tags      []string `json:",omitempty" yaml:",omitempty"`
+}
+
+type BuiltImageSourceLocal struct {
+	Type string // always set to 'local'
+	Path string
+}
+
+type ResolvedImageSourceURL struct {
+	Type string // always set to 'resolved'
+	URL  string
+	Tag  string
+}
+
+type TaggedImageMeta struct {
+	Type string // always set to 'tagged'
+	Tags []string
+}
+
+type PreresolvedImageSourceURL struct {
+	Type string // always set to 'preresolved'
+	URL  string
+	Tag  string `json:",omitempty" yaml:",omitempty"`
+}
+
+func (BuiltImageSourceGit) meta()       {}
+func (BuiltImageSourceLocal) meta()     {}
+func (ResolvedImageSourceURL) meta()    {}
+func (TaggedImageMeta) meta()           {}
+func (PreresolvedImageSourceURL) meta() {}
+
+func metasHistory(metas string) ([]Meta, error) {
```

Comment:
```suggestion
func parseMetas(metas string) ([]Meta, error) {
```

It would be great if "Meta" had a better metaphor... and what I'm seeing here is an attempt to suggest one: something less reliable than "provenance" but tells a story from whence this image came: a history.

However, in _this_ context, this function is effectively taking the unmarshalling just one more step to include these `Meta` types. So, the intention of the function actually gets a bit blurred.

Another angle to "justify" this suggestion is that functions often _tend to_ work better as verb[-phrases] than nouns.

"Parse" rather than "unmarshall" because the function accepts an unstructured input and _produces_ a structured output, rather than stuffing a structure. 🤷🏻